engine/src/main/java/org/hibernate/validator/internal/engine/cascading/LegacyOptionalValueExtractor.java

```diff
@@ -17,6 +18,7 @@
  * @author Gunnar Morling
  */
 // TODO should we keep that, or only support {@code Optional<@Valid Foo>}.
+@UnwrapByDefault
```

Comment:
Why is that actually? As `Optional` has a type parameter, constraints applying to the wrapped value should be given on the type argument, hence no need for default-unwrapping for element-level constraints. Same for the non-legacy extractor.